test/tests/functional/pbs_qstat_formats.py

```diff
@@ -409,6 +409,36 @@ def test_qstat_json_valid_multiple_jobs(self):
         except ValueError:
             self.assertTrue(False)
 
+    def test_qstat_json_valid_multiple_jobs_p(self):
+        """
+        Test json output of qstat -f is in valid format when array jobs are
+        queried and make sure that attributes are displayed with `p` option
+        """
+        j = Job(TEST_USER)
+        j.set_sleep_time(100)
+        jid = self.server.submit(j)
+        jid2 = self.server.submit(j)
+        jid3 = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': "R"}, id=jid)
+        self.server.expect(JOB, {'job_state': "R"}, id=jid2)
+        self.server.expect(JOB, {'job_state': "R"}, id=jid3)
+        qstat_cmd_json = os.path.join(self.server.pbs_conf['PBS_EXEC'], 'bin',
+                                      'qstat') + ' -fp -F json '
```

Comment:
just curious, why are you passing -p to qstat in this test?

Reply:
If `-p` is passed, then only the Resource_List is requested. An attribute with type resource list has to be the last attribute in order to hit the bug.

Reply:
Got it. Can you please add that as a comment in the test?